src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
So, if ReceiveHeadersTimeout is 30 seconds (which was the documented default for WinHttpHandler):

* Create request
* Send the GET and request headers
* Wait unbounded
* libcurl receives a header
* The CurlReceiveHeadersCallback fires
* If the data rate stays below 1 byte per second for 30 seconds, abort, otherwise read forever.
    * Except we already read, since curl already told us how big stuff is.  So maybe it's the timeout for the next header?
    * Repeat for every header.

If it makes sense to you and David I'll accept it.  I'm just confused between the name of the functions in the code (and their associated comments), the curl documentation, and what the timeout actually covers.